pkg/trace/sampler/prioritysampler.go

```diff
@@ -110,30 +120,55 @@ func (s *PrioritySampler) Sample(trace pb.Trace, root *pb.Span, env string, clie
 		return sampled
 	}
 
-	signature := s.catalog.register(ServiceSignature{root.Service, env})
+	signature := s.catalog.register(ServiceSignature{Name: root.Service, Env: env})
 
 	// Update sampler state by counting this trace
-	s.Sampler.Backend.CountSignature(signature)
+	s.CountSignature(root, signature)
 
 	if sampled {
 		rate := s.applyRate(sampled, root, signature)
-		s.Sampler.Backend.CountSample()
-		// adjust sig score with the expected P0 count for that sig
-		// adjusting this score only matters for root spans
-		if clientDroppedP0s && rate > 0 && rate < 1 {
-			// removing 1 to not count twice the P1 chunk
-			weight := 1/rate - 1
-			s.Sampler.Backend.CountWeightedSig(signature, weight)
-		}
+		s.CountSampled(root, clientDroppedP0s, signature, rate)
 	}
 	return sampled
 }
 
+func (s *PrioritySampler) CountSignature(root *pb.Span, signature Signature) {
+	s.localRates.Backend.CountSignature(signature)
+
+	// remoteRates only considers root spans
+	if s.remoteRates == nil || root.ParentID != 0 {
```

Comment:
nit: I think it would be clearer to write the condition to update the remote rates `s.remoteRates != nil && root.ParentID == 0` than its oppostie